src/index.d.ts

```diff
@@ -52,17 +51,32 @@ export interface JsonRpcFailure<T> extends JsonRpcResponseBase {
 export type JsonRpcResponse<T> = JsonRpcSuccess<T> | JsonRpcFailure<T>
 
 export type JsonRpcEngineEndCallback = (error?: JsonRpcError<unknown>) => void;
+
+type ReturnHandlerCallback = (done: (error?: Error) => void) => void
+
 export type JsonRpcEngineNextCallback = (
-  returnFlightCallback?: (done: () => void) => void,
+  returnHandlerCallback?: ReturnHandlerCallback,
```

Comment:
This was one of the errors. You can actually pass errors to the return handler callbacks in order to cause the JSON-RPC request to return an error.